test/unit/model/LaunchServiceTest.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA
+ */
+
+declare(strict_types=1);
+
+namespace oat\remoteProctoring\test\unit\model;
+
+use GuzzleHttp\Psr7\Request;
+use oat\generis\test\TestCase;
+use oat\remoteProctoring\model\LaunchService;
+use oat\remoteProctoring\model\signature\exception\SignatureException;
+use oat\remoteProctoring\model\signature\SignatureMethod;
+use Prophecy\Argument;
+use Psr\Http\Message\RequestInterface;
+
+class LaunchServiceTest extends TestCase
+{
+    /**
+     * @var LaunchService
+     */
+    private $subject;
+
+    protected function setUp(): void
+    {
+        $this->subject = new LaunchService([LaunchService::OPTION_SIGNATURE_METHOD => $this->getSigner()]);
+    }
+
+    public function testValidateRequest()
+    {
+        $this->assertNull($this->subject->validateRequest($this->getRequest('https://google.com&validSignature')));
+
+        $this->expectException(SignatureException::class);
+        $this->subject->validateRequest($this->getRequest('https://google.com&invalidSignature'));
+    }
```

Comment:
There is not need to mock a dependency behavior here. If the dependency internal behavior changes this test will not be affected (cause this is not an integration test).

```suggestion
    public function testValidateRequest()
    {
        $this->assertNull($this->subject->validateRequest($this->getRequest('https://google.com&validSignature')));
    }
```
The method `getSigner` can also be shortened.

Reply:
To be honest I don't like this approach @gabrielfs7 . I would recommend keeping this as it is in order to achieve more code readability. 

It's much easier to understand what is going on in this test when you have each operation separated. 

_IMO_ Test is kind of document where your method is actually described. Simplification of this description would be desired. This is not production code. This is only for developers so we not gaining or losing performance by these actions.

Reply:
@bartlomiejmarszal  If you check my other comment I suggested to split in multiple tests. This is just part of the idea proposed. You should test either the negative or positive in a unit, but not both. As you said, it is a document and should be easy to read and do not mix things.

Reply:
guys, i would like to ask you to ask this change only if this is mandatory,  otherwise we can have a technical dept ticket that we will handle in the next sprints

Reply:
This was mandatory, cause it was testing external usage of the class, so it was an invalid test, which should be removed. @boajer fixed. Thanks!! :)